src/core/unit.c

```diff
@@ -4853,7 +4853,6 @@ bool unit_is_pristine(Unit *u) {
         return IN_SET(u->load_state, UNIT_NOT_FOUND, UNIT_LOADED) &&
                !u->fragment_path &&
                !u->source_path &&
-               strv_isempty(u->dropin_paths) &&
                !u->job &&
                !u->merged_into;
```

Comment:
can you add a comment here, mentioning that we explicitly do *not* check for drop-in paths, and why?